cli/analyzer/windows/registry.go

```diff
@@ -0,0 +1,430 @@
+	// registry under /var/snapshot/Windows/System32/config/SOFTWARE, this should
+	// extract /var/snapshot mount path
+	mountPath := registryFilePath
+	for _, defaultRootPath := range defaultRegistryRootPaths {
+		mountPath = strings.TrimSuffix(mountPath, defaultRootPath)
+	}
+
+	return &Registry{
+		mountPath:   mountPath,
+		softwareReg: softwareReg,
+		cleanup:     registryFile.Close,
+		logger:      logger,
+	}, nil
+}
+
+// Close needs to be called when done to free up resources. Registry is not
+// usable once closed.
+func (r *Registry) Close() error {
+	if err := r.cleanup(); err != nil {
+		return fmt.Errorf("unable to close registry: %w", err)
+	}
+	return nil
+}
+
+// GetPlatform returns OS-specific data from the registry.
+func (r *Registry) GetPlatform() (map[string]string, error) {
+	// Open key to fetch operating system version and configuration data
+	platformKey, err := openKey(r.softwareReg, "Microsoft/Windows NT/CurrentVersion")
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract all platform data from the registry
+	platform := getValuesMap(platformKey)
+
+	// Strip information about the product key hash
+	delete(platform, "DigitalProductId")
+	delete(platform, "DigitalProductId4")
+
+	return platform, nil
+}
+
+// GetUpdates returns a slice of all installed system updates from the registry.
+func (r *Registry) GetUpdates() ([]string, error) {
+	// Open key to fetch CBS data about packages (updates and components)
+	packagesKey, err := openKey(r.softwareReg, "Microsoft/Windows/CurrentVersion/Component Based Servicing/Packages")
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract all updates from installed packages
+	updates := map[string]string{}
+	updateRegex := regexp.MustCompile("KB[0-9]{7,}")
+	for _, pkgKey := range packagesKey.Subkeys() {
+		pkgName := pkgKey.Name()
+		pkgValues := getValuesMap(pkgKey)
+
+		// Ignore packages that were not installed as system components or via updates
+		_, isComponent := pkgValues["InstallClient"]
+		_, isUpdate := pkgValues["UpdateAgentLCU"]
+		if !isComponent && !isUpdate {
+			continue
+		}
+
+		// Install location value for a given package can contain update identifier such
+		// as "C:\Windows\CbsTemp\31075171_2144217839\Windows10.0-KB5032189-x64.cab\"
+		if location, ok := pkgValues["InstallLocation"]; ok {
+			if kb := updateRegex.FindString(location); kb != "" {
+				updates[kb] = kb
+			}
+		}
+
+		// If the installed package contains state value, it indicates a potential system
+		// update. We are only curious about "112" state code which translates to
+		// successful package installation. When this is the case, package registry key
+		// contains update identifier such as "Package_10_for_KB5011048..."
+		if state, ok := pkgValues["CurrentState"]; ok && state == "112" {
+			if kb := updateRegex.FindString(pkgName); kb != "" {
+				updates[kb] = kb
+			}
+		}
+	}
+
+	return utils.StringKeyMapToArray(updates), nil
```

Comment:
I'd use https://github.com/openclarity/vmclarity/blob/main/core/to/to.go#L54 instead.

Reply:
got my rebases wrong, will do